src/components/data/PathListAutomation.js

```diff
@@ -0,0 +1,405 @@
+                setInfoTypeError(e);
+            },
+        },
+    });
+
+    const [createPathListFile, { status }] = useMutation(
+        ({ submission }) => pathListCreator(submission),
+        {
+            onSuccess: (data, { onSuccess }) => {
+                onSuccess();
+                setCreatePathListError(null);
+                onCreatePathList(data.file.id, data.file.path);
+            },
+            onError: (error, { onError }) => {
+                let errMsg = null;
+                const err_code = getErrorCode(error);
+                switch (err_code) {
+                    case ERROR_CODES.ERR_EXISTS:
+                        errMsg = t("pathListExistError");
+                        onError(err_code);
+                        break;
+
+                    case ERROR_CODES.ERR_NOT_FOUND:
+                        errMsg = t("pathListNoMatchError");
+                        onError();
+                        break;
+
+                    default:
+                        errMsg = t("infoTypeFetchError");
+                        onError();
+                }
+                setCreatePathListError(error);
+                setErrorMsg(errMsg);
+            },
+        }
+    );
+
+    useEffect(() => {
+        if (!infoTypesCache || infoTypesCache.length === 0) {
+            setInfoTypesQueryEnabled(true);
+        } else {
+            if (infoTypes === null || infoTypes.length === 0) {
+                setInfoTypes(infoTypesCache.types);
+            }
+        }
+    }, [infoTypes, infoTypesCache]);
+
+    const handleToggle = (value) => () => {
+        const currentIndex = selectedInfoTypes.indexOf(value);
+        const newChecked = [...selectedInfoTypes];
+
+        if (currentIndex === -1) {
+            newChecked.push(value);
+        } else {
+            newChecked.splice(currentIndex, 1);
+        }
+
+        setSelectedInfoTypes(newChecked);
+    };
+
+    const handlePathListCreation = (values, actions) => {
+        const { selectedPaths, pattern, dest, foldersOnly } = values;
+        actions.setSubmitting(true);
+
+        const submission = {
+            paths: selectedPaths,
+            dest,
+            pattern,
+            foldersOnly: foldersOnly || false,
+            recursive: true,
+            requestedInfoType,
+            selectedInfoTypes,
+        };
+        const onSuccess = () => {
+            actions.setSubmitting(false);
+        };
+        const onError = (err_code) => {
+            if (err_code === ERROR_CODES.ERR_EXISTS) {
+                actions.setFieldError("dest", t("pathListExistError"));
+            }
+            actions.setSubmitting(false);
+        };
+        createPathListFile({
+            submission,
+            onSuccess,
+            onError,
+        });
+    };
+
+    const onSwitchChange = (setFieldValue, fieldName) => (event, checked) => {
+        setFieldValue(fieldName, checked);
+    };
+
+    const FormSwitch = ({
```

Comment:
I think `FormCheckBox` from `ui-lib` would also work well here if you wanted less code to manage in this component.  Otherwise, this might be good to add to `ui-lib`.

Reply:
Ah ha!  I thought I had seen this somewhere but only checked `ui-lib`.  Good catch!